module/VuFind/src/VuFind/Marc/Serialization/Iso2709.php

```diff
@@ -176,4 +243,47 @@ public static function toString(string $leader, array $fields): string
 
         return $leader . $directory . $data;
     }
+
+    /**
+     * Open a collection file
+     *
+     * @param string $file File name
+     *
+     * @return void
+     */
+    public function openCollectionFile(string $file): void
+    {
+        if (false === ($this->file = fopen($file, 'rb'))) {
+            throw new \Exception("Cannot open file '$file' for reading");
+        }
+    }
+
+    /**
+     * Rewind the collection file
+     *
+     * @return void;
+     */
+    public function rewind(): void
+    {
+        rewind($this->file);
```

Comment:
Should we add some validation here and in getNextRecord to throw an exception if these get called when no file has been opened yet?

Reply:
I first decided against it as the usage is protected by MarcCollectionFile, but now I realized that there's nothing stopping someone from instantiating the class standalone, so I added some checks.

Reply:
Thanks!